### PR TITLE
Multiedi

### PR DESCRIPTION
**Pull request by andrewfraser73**
change supplier selection on edit to a datagrid based on possible edi suppliers where you can edit the email

view of edi suppliers is a bit slow but still faster than individual lookups.

### CODE REVIEW

File: src/Facade/ResourceBuilders/EdiSupplierResourceBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        public EdiSupplierResource BuildResource(EdiSupplier model, IEnumerable<string> claims)
+        {
+            return new EdiSupplierResource { 
+                SupplierId = model.SupplierId, 
+                SupplierName = model.SupplierName, 
+                VendorManager = model.VendorManager, 
+                VendorManangerName = model.VendorManangerName, 
+                EdiEmailAddress = model.EdiEmailAddress, 
+                NumOrders=model.NumOrders
```
Comment (lewisrenfrew) [inline src/Facade/ResourceBuilders/EdiSupplierResourceBuilder.cs:20]: missing space after =

File: src/Persistence.LinnApps/ServiceDbContext.cs [hunk 2]
```diff
@@ -1443,6 +1446,18 @@ private void BuildEdiOrders(ModelBuilder builder)
...
+        {
+            var entity = builder.Entity<EdiSupplier>().ToTable("EDI_PENDING_VIEW");
+            entity.HasKey(e => e.SupplierId);
+            entity.Property(d => d.SupplierId).HasColumnName("SUPPLIER_ID");
+            entity.Property(d => d.SupplierName).HasColumnName("SUPPLIER_NAME").HasColumnType("VARCHAR2");
+            entity.Property(d => d.VendorManager).HasColumnName("VENDOR_MANAGER").HasColumnType("VARCHAR2");
+            entity.Property(d => d.VendorManangerName).HasColumnName("USER_NAME").HasColumnType("VARCHAR2");
+            entity.Property(d => d.EdiEmailAddress).HasColumnName("EDI_EMAIL_ADDRESS").HasColumnType("VARCHAR2");
+            entity.Property(d => d.NumOrders).HasColumnName("NUM_ORDERS");
+		}
```
Comment (lewisrenfrew) [inline src/Persistence.LinnApps/ServiceDbContext.cs:1459]: weird indentation? Or is that just github

File: src/Resources/EdiSupplierResource.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿namespace Linn.Purchasing.Resources
+{
+    using Linn.Common.Resources;
+
+    public class EdiSupplierResource : HypermediaResource
```
Comment (lewisrenfrew) [inline src/Resources/EdiSupplierResource.cs:5]: Shouldn't need to implement HyperMediaResource if you don't use the Links list

File: src/Service.Host/client/src/components/EdiOrders.js [hunk 4]
```diff
@@ -62,23 +124,28 @@ function EdiOrder() {
...
-    const [altEmail, setAltEmail] = useState('');
-    const [additionalEmail, setAdditionalEmail] = useState('');
     const [additionalText, setAdditionalText] = useState('');
 
     const sendEdiUrl = utilities.getHref(applicationState, 'edi');
 
-    const handleSupplierChange = selectedsupplier => {
-        setSupplier(selectedsupplier);
+    const handleSelectRow = selected => {
+        const newRows = rows.map(r =>
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/EdiOrders.js:132]: `rows.map(r => {...r, selected: selected.includes(r.id) }`

is more concises

**Review by lewisrenfrew — APPROVED** [review]

File: src/Persistence.LinnApps/ServiceDbContext.cs (hunk 2 above)
Comment (andrewfraser73) [reply src/Persistence.LinnApps/ServiceDbContext.cs:1459]: weird looks okay in visual studio

**Pull request merged**